dag_bakery/callbacks/context_callback.py

```diff
@@ -0,0 +1,69 @@
+from abc import ABC, abstractmethod
+from typing import List, TypeVar, Any, Optional, Generic
+
+from airflow import DAG
+from pydantic import BaseModel
+
+
+class ContextCallback(ABC):
+    @abstractmethod
+    def callback(self, context: dict) -> Any:
+        pass
+
+    def __call__(self, context: dict):
+        return self.callback(context)
+
+
+ContextCallbackType = TypeVar("ContextCallbackType", bound=ContextCallback)
+
+
+def chain_callbacks(ctx_callbacks: List[ContextCallbackType]):
+    def execute(context: dict):
+        for ctx_callback in ctx_callbacks:
+            ctx_callback(context)
+
+    return execute
+
+
+class CallbacksConfig(BaseModel, Generic[ContextCallbackType]):
+    task_success_callbacks: Optional[List[ContextCallbackType]] = None
+    task_failure_callbacks: Optional[List[ContextCallbackType]] = None
+    dag_success_callbacks: Optional[List[ContextCallbackType]] = None
+    dag_failure_callbacks: Optional[List[ContextCallbackType]] = None
+
+
+def set_task_success_callbacks(dag: DAG, callbacks: List[ContextCallbackType]):
```

Comment:
These four functions are the same (pairwise). Can we just have 2 somehow?

Reply:
They could 🤔 , but I would like to give the freedom to anyone to use the required one.
We could have a 3rd function that receive the `property` to be defined as a param, but it would be a bit overplate.

Reply:
I mean what they do, not what they are named. They all do something to the tasks or dag, and add a list doing something to it. The only difference is in what property of the dag or task they modify

Reply:
Since there doesn't seem to be any way that is not too cumbersome, let's leave it like this. Maybe we come up with something later, in any case it's just 3 lines 😄